src/Init/Prelude.lean

```diff
@@ -3721,6 +3721,12 @@ abbrev nameLitKind : SyntaxNodeKind := `name
 /-- `fieldIdx` is the node kind of projection indices like the `2` in `x.2`. -/
 abbrev fieldIdxKind : SyntaxNodeKind := `fieldIdx
 
+/--
+`macroScope` is the node kind of macro scopes, which are "invisible tokens"
+which capture the hygiene information at the current point without parsing anything.
+-/
```

Comment:
```suggestion
They can be used to generate identifiers (with `Lean.TSyntax.mkIdent`) as if they were
introduced by the calling context, not the called macro. -/
```
or something to that effect

Reply:
Perhaps `TSyntax` is not the ideal namespace for this function even if it is convenient. Or maybe it needs a more explicit name like `mkIdentFromMacroScope`

Reply:
I was a bit surprised to see that `getNat` etc are in the `TSyntax` namespace instead of `NatLit`, but I guess it doesn't work if you use that namespace?

Reply:
It only works if you have an explicit type annotation `(stx : NumLit)`, but not with `$stx:num`. It's not ideal, no.

Reply:
I went with `HygieneInfo.mkIdent`, which is a compromise which sometimes works with dot notation and reads ok even if you can't use dot notation.